From adb896c080670159b4a2455e3da57ebc0b93b2fd Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Wed, 10 Dec 2014 13:07:56 +0100
Subject: [PATCH 0109/2241] fix: i2c: add suspend/resume support

	- There is no need to implement subroutine for suspend since there is no
	data to store before suspending
	- Bug fix for: 'SYSTEMSW-1425 I2C - suspend/resume is not supported'

Change-Id: Ie927b93e65fbfb28cc6c7ef977dc327b33263293
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/15256
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/15365
Reviewed-on: http://vgitil04.il.marvell.com:8080/27485
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
---
 drivers/i2c/busses/i2c-mv64xxx.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/i2c/busses/i2c-mv64xxx.c b/drivers/i2c/busses/i2c-mv64xxx.c
index c68bddc..ef7b35d 100644
--- a/drivers/i2c/busses/i2c-mv64xxx.c
+++ b/drivers/i2c/busses/i2c-mv64xxx.c
@@ -994,9 +994,21 @@ mv64xxx_i2c_remove(struct platform_device *dev)
 	return 0;
 }
 
+static int mv64xxx_i2c_resume(struct platform_device *dev)
+{
+	struct mv64xxx_i2c_data *drv_data = platform_get_drvdata(dev);
+
+	mv64xxx_i2c_hw_init(drv_data);
+
+	return 0;
+}
+
 static struct platform_driver mv64xxx_i2c_driver = {
 	.probe	= mv64xxx_i2c_probe,
 	.remove	= mv64xxx_i2c_remove,
+#ifdef CONFIG_PM
+	.resume = mv64xxx_i2c_resume,
+#endif
 	.driver	= {
 		.name	= MV64XXX_I2C_CTLR_NAME,
 		.of_match_table = mv64xxx_i2c_of_match_table,
-- 
2.7.4

